src/swarm-tui/Swarm/TUI/Controller/Util.hs

```diff
@@ -117,3 +157,41 @@ resetViewport :: ViewportScroll Name -> EventM Name AppState ()
 resetViewport n = do
   vScrollToBeginning n
   hScrollToBeginning n
+
+-- | Modifies the game state using a fused-effect state action.
+zoomGameState :: (MonadState AppState m, MonadIO m) => Fused.StateC GameState (TimeIOC (Fused.LiftC IO)) a -> m a
+zoomGameState f = do
+  gs <- use gameState
+  (gs', a) <- liftIO (Fused.runM (runTimeIO (Fused.runState gs f)))
+  gameState .= gs'
+  return a
+
+onlyCreative :: (MonadState AppState m) => m () -> m ()
+onlyCreative a = do
+  c <- use $ gameState . creativeMode
+  when c a
+
+-- | Create a list of handlers with embedding events and using pattern matching.
+allHandlers ::
+  (Ord e2, Enum e1, Bounded e1) =>
+  (e1 -> e2) ->
+  (e1 -> (Text, EventM Name AppState ())) ->
+  [KeyEventHandler e2 (EventM Name AppState)]
+allHandlers eEmbed f = map handleEvent1 [minBound .. maxBound]
```

Comment:
For `[minBound .. maxBound]` use `Swarm.Util.listEnums`.

Reply:
Thanks for the tip, I suspected this was a common pattern already. 👍

Reply:
BREAKING NEWS: use `enumerate` instead. :facepalm:  #2003

Reply:
Rebased and done again! ✅